src/osd/recovery_types.h

```diff
@@ -13,32 +13,74 @@
  * Represents the objects in a range [begin, end)
  *
  * Possible states:
- * 1) begin == end == hobject_t() indicates the the interval is unpopulated
- * 2) Else, objects contains all objects in [begin, end)
+ * 1) Unpopulated - only interval's range is set
+ * 2) Populatd - objects contains all objects in [begin, end)
  */
+
 struct BackfillInterval {
+private:
   // info about a backfill interval on a peer
   eversion_t version; /// version at which the scan occurred
   std::map<hobject_t,eversion_t> objects;
-  hobject_t begin;
-  hobject_t end;
+  hobject_t begin; /// object to start populating the interval from
+  hobject_t end;   /// object to start populating the interval to
+  bool populated = false;
+
+public:
+  BackfillInterval(hobject_t begin);
+
```

Comment:
Same here, it would probably be worth adding a comment explaining what this constructor is meant to do.  I think there's also an implicit invariant that end >= begin that you'll want to mention.